nixos/modules/services/web-apps/birdsitelive.nix

```
@@ -0,0 +1,625 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+let
+  cfg = config.services.birdsitelive;
+  db = cfg.settings.Db;
+
+  isAbsolutePath = v: isString v && substring 0 1 v == "/";
+  isSecret = v: isAttrs v && v ? _secret && isAbsolutePath v._secret;
+
+  secret = mkOptionType {
+    name = "secret";
+    description = "secret value";
+    descriptionClass = "noun";
+    check = isSecret;
+    nestedTypes = {
+      _secret = types.str;
+    };
+  };
+
+  sha256 = builtins.hashString "sha256";
+
+  toJsonSec = let
+    concatItems = concatStringsSep ",";
+    toJsonSec' = { }@args: v:
+      if isAttrs v
+        then if v ? _secret
+          then if isAbsolutePath v._secret
+            then ''"${sha256 v._secret}"''
+            else abort "Invalid secret path (_secret = ${v._secret})"
+          else "{${concatItems (mapAttrsToList (key: val: ''"${key}":${toJsonSec' args val}'') v)}}"
+        else if isList v
+          then "[${concatItems (map (toJsonSec' args) v)}]"
+          else builtins.toJSON v;
+  in toJsonSec' { };
+
+  configFile = pkgs.runCommand "appsettings.json" {
+    nativeBuildInputs = with pkgs; [ jq ];
+    value = toJsonSec cfg.settings;
+    passAsFile = [ "value" ];
+  } ''jq . "$valuePath" >$out'';
+
+  secretPaths = catAttrs "_secret" (collect isSecret cfg.settings);
+
+  writeShell = { name, text, runtimeInputs ? [ ] }:
+    pkgs.writeShellApplication { inherit name text runtimeInputs; } + "/bin/${name}";
+
+  configScript = writeShell {
+    name = "birdsitelive-config";
+    runtimeInputs = with pkgs; [ coreutils replace-secret ];
+    text = ''
+      cd "$RUNTIME_DIRECTORY"
+      tmp="$(mktemp appsettings.json.XXXXXXXXXX)"
+      trap 'rm -f "$tmp"' INT TERM HUP EXIT
+
+      cat ${escapeShellArg configFile} >"$tmp"
+      ${concatMapStrings (file: ''
+        replace-secret ${escapeShellArgs [ (sha256 file) file ]} "$tmp"
+      '') secretPaths}
+
+      chown ${escapeShellArg cfg.user}:${escapeShellArg cfg.group} "$tmp"
+      chmod 0400 "$tmp"
+      mv "$tmp" appsettings.json
+    '';
+  };
+
+  managerConfigScript = writeShell {
+    name = "bslmanager-config";
+    runtimeInputs = with pkgs; [ coreutils jq ];
+    text = ''
+      cd "$RUNTIME_DIRECTORY"
+      tmp="$(mktemp ManagerSettings.json.XXXXXXXXXX)"
+      trap 'rm -f "$tmp"' INT TERM HUP EXIT
+
+      ${pkgs.jq}/bin/jq '{
```

```suggestion
      jq '{
```
if runtimeInputs is working correct we don't need this, right?